src/Agent.Sdk/CommandPlugin.cs

```diff
@@ -48,11 +48,14 @@ public AgentCommandPluginExecutionContext()
             this.Properties = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
             this.Variables = new Dictionary<string, VariableValue>(StringComparer.OrdinalIgnoreCase);
 
-#if OS_WINDOWS
-            this.ContainerPathMappings = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
-#else
-            this.ContainerPathMappings = new Dictionary<string, string>();
-#endif
+            if (PlatformUtil.RunningOnOS == PlatformUtil.OS.Windows)
+            {
+                this.ContainerPathMappings = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            }
+            else
+            {
+                this.ContainerPathMappings = new Dictionary<string, string>();
+            }
```

Comment:
not a huge deal, but it might be better to use the following pattern for these types of things:

```
StringComparer comparer = StringComparer.Ordinal;
if (PlatformUtil.RunningOnOS == PlatformUtil.OS.Windows)
{
    comparer  = StringComparer.OrdinalIgnoreCase;
}
this.ContainerPathMappings = new Dictionary<string,string>(comparer);
             
```

Reply:
Agreed, thanks. I started doing something like that later, and will fix this.